pkg/controller/status/status.go

```diff
@@ -162,13 +162,13 @@ func (m *statusManager) updateStatus() {
 		// We've collected knowledge about the current state of the objects we're monitoring.
 		// Now, use that to update the TigeraStatus object for this manager.
 		if m.IsAvailable() {
-			m.setAvailable("All objects available", "")
+			m.setAvailable(string(operator.AllObjectsAvailable), "All objects available")
```

Comment:
I would prefer if the setAvailable method only accepted TigeraStatusReasons:
```
func (m *statusManager) setAvailable(reason operator.TigeraStatusReason, msg string) {
```
Similar for the other methods. That way, you can never enter the wrong string in the reason field.

Reply:
And same for the StatusManager interface.

Reply:
Sure Rene. i will make this changes once i finish this implementation for all the CR's
since
- Signature of SetAvailable and setProgressing can be changed as it is harded but for SetDegarded, the reason is passed across from all the CR controllers. 
- Few Unit test cases in others CRs are based on the current free form reason string which i am fixing along with the respective CR changes

Reply:
SGTM!